src/vma/proto/dst_entry_udp.cpp

```diff
@@ -429,8 +415,36 @@ void dst_entry_udp::init_sge()
 #endif /* DEFINED_TSO */
 }
 
+ssize_t dst_entry_udp::check_payload_size(const iovec* p_iov, ssize_t sz_iov)
+{
+	// Calc user data payload size
+	ssize_t sz_data_payload = 0;
+	for (ssize_t i = 0; i < sz_iov; i++) {
+		// Imitate Kernel behaviour.
+		if (unlikely(!p_iov[i].iov_base) && unlikely(p_iov[i].iov_len)) {
+			errno = EFAULT;
+			return -1;
+		}
+
+		sz_data_payload += p_iov[i].iov_len;
+	}
+
+	if (unlikely(sz_data_payload > 65507)) { // See comment in sockinfo_udp::tx
+		dst_udp_logfunc("sz_data_payload=%d, to_port=%d, local_port=%d", sz_data_payload, ntohs(m_dst_port), ntohs(m_src_port));
+		dst_udp_logfunc("sz_data_payload=%d exceeds max of 64KB", sz_data_payload);
+		errno = EMSGSIZE;
+		return -1;
+	}
```

Comment:
Should not it be covered by https://github.com/Mellanox/libvma/blob/master/src/vma/sock/sockinfo_udp.cpp#L1542-L1556 ?

Reply:
This check is now different. See the new sockinfo_udp::tx. It was checking only the first element which is not enough and incorrect, since the first element can be null.

Reply:
I am not sure that it can be done in fast_send() because we would like to handle wrong cases by OS (go to tx_packet_to_os). OS should process these cases including correct errno. In this case we can be aligned with kernel behavior.

Reply:
These checks were already done in fast_send but now its just move to a function to cover more cases and done more precisely.